### ISSUE TITLE
Menu list extends beyond screen

### ISSUE THREAD

**Issue by Mailleweaver** [notepad-plus-plus/notepad-plus-plus#12815]
### Description of the Issue
If your main monitor has a vertical resolution greater than a secondary monitor, long Notepad++ menu lists (like the Language menu) will extend beyond the screen when Notepad is being displayed on a secondary monitor. This prevents scrolling of the list and selection of the bottom items.

### Steps to Reproduce the Issue
1. Have multiple monitors with the larger one set as the primary.
2. Put Notepad++ on a secondary monitor.
3. Open a menu that is longer than the secondary screen's resolution.

### Expected Behavior
The list for the opened menu will be shortened to fit the monitor where the program is being displayed.

### Actual Behavior
The list fits the main monitor's resolution instead of the monitor it's being displayed on.

### Debug Information
Notepad++ v8.4.8   (32-bit)
Build time : Dec 24 2022 - 19:30:12
Path : C:\Program Files (x86)\Notepad++\notepad++.exe
Command Line : 
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 10 Home (64-bit) 
OS Version : 22H2
OS Build : 19045.2486
Current ANSI codepage : 1252
Plugins : 
    mimeTools (2.9)
    NppConverter (4.5)
    NppExport (0.4)

![image](https://user-images.githubusercontent.com/1341522/213123056-ecc90cb8-6111-44c8-86f3-b4a01ff49b43.png)

**Comment by Mailleweaver** [comment]
Language menu.
You could be right. If there is a rectangle enclosing the desktop area that
would contain all monitors, then it wouldn't matter which one is the
primary monitor since that rectangle would still give an un-displayed
region for the menu to extend into.
Maybe the menu list should be contained within the program's window area
rather than attempting to fit the entire display area. That would be a
reasonable and intuitive boundary.

**Comment by alankilborn** [comment]
>>Which menu do you have dropped down in the screenshot (can't quite make it out)?

>Language menu.

Ah, and you have _Settings_ > _Preferences..._ > _Language_ > _Make language menu compact_ unchecked.

**Comment by Mailleweaver** [comment]
Yeah, it looks like it. I don't remember doing that, but I must have since my other computer has it checked. I wondered why I didn't have to scroll the menu on the other computer.

Though, even with it not being compact, it should still not extend beyond both the screen and the window.

**Comment by alankilborn** [comment]
I didn't notice it ever with the _Language_ menu, as I have the default "compact" setting enabled.

I noticed it when using the _Macro_ menu when I have "too many" macros.
Side note: With N++ 8.4.8, I don't have this problem as I have reorganized my _Macro_ menu contents using the new submenu capability.

**Comment by alankilborn** [comment]
Possibly relevant/related:  https://github.com/notepad-plus-plus/notepad-plus-plus/issues/8401#issuecomment-642319081

**Comment by Mailleweaver** [comment]
I'm a very basic user. The only feature I use that's more complicated than Find and Replace is the language editor to color code and code block the simple scripts I write for work. I don't understand macros.

Yeah, that looks like the second user had the same issue as me. And as for the main point of that thread, I'd agree that scrolling with the wheel would be better than having to click the down arrow. Every single time I've had to scroll the Language menu, I tried scrolling with the wheel first because that's what is natural.

**Comment by alankilborn** [comment]
>I tried scrolling with the wheel first because that's what is natural.

Agreed.  Anything that presents a dropdown where the drop isn't long enough to show all the choices at once should have this type of capability.  A recent example of where this was sorely lacking for a long time was in the _Find_ window's _Find what_ dropdown edit box; recently, due to my urging, that was added.
